tests/integ/modin/test_telemetry.py

```diff
@@ -419,22 +420,33 @@ def test_telemetry_getitem_setitem():
 
 
 @pytest.mark.parametrize(
-    "name, expected_func_name, method, expected_query_count",
+    "name, expected_func_name, method, expected_query_count, method_implemented",
     [
         # __repr__ is an extension method, so the class name is shown only once.
-        ["__repr__", "DataFrame.__repr__", lambda df: df.__repr__(), 1],
+        ["__repr__", "DataFrame.__repr__", lambda df: df.__repr__(), 1, True],
         # __iter__ was defined on the DataFrame class, so it is shown twice.
-        ["__iter__", "DataFrame.DataFrame.__iter__", lambda df: df.__iter__(), 0],
+        ["__iter__", "DataFrame.DataFrame.__iter__", lambda df: df.__iter__(), 0, True],
+        [
+            "__dataframe__",
+            "DataFrame.__dataframe__",
+            lambda df: df.__dataframe__(),
+            0,
+            False,
+        ],
     ],
 )
 def test_telemetry_private_method(
-    name, expected_func_name, method, expected_query_count
+    name, expected_func_name, method, expected_query_count, method_implemented
 ):
     df = pd.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]})
     # Clear connector telemetry client buffer to avoid flush triggered by the next API call, ensuring log extraction.
     df._query_compiler._modin_frame.ordered_dataframe.session._conn._telemetry_client.telemetry.send_batch()
 
-    with SqlCounter(query_count=expected_query_count):
+    with SqlCounter(query_count=expected_query_count), (
+        contextlib.nullcontext()
```

Comment:
What's the purpose of a context manager here?

Reply:
`pytest.raises()` is a context manager that checks for an error. We use it when we expect a `NotImplementedError` here. `nullcontext()` gives a neat way to conditionally add the `pytest.raises()` context manager when we expect an error, and to do nothing otherwise.

Reply:
I see, thanks!